lib/translators/javascript.js

```diff
@@ -10,7 +12,8 @@ function translateJavascript(copy, gettext) {
     needle = match[0];
     msgid = match[1] || match[2];
     bookend = match[1] ? '"' : '\'';
-    replacement = bookend + gettext(msgid) + bookend;
+    var escapedString = lodash.escape(gettext(msgid));
+    replacement = bookend + escapedString + bookend;
```

Comment:
I don't think it makes sense to have this library substitute an unescaped msgid with an escaped version. That's way too confusing for the user.

The problem that this PR is trying to address is strings that contain quotation marks. I think a simpler solution to those would be to embed the `unescape()` call in the outputted JS.